features/extensions/apigee/setup.py

```diff
@@ -0,0 +1,312 @@
+    jazz_userpass_list = ''.join(list(jazz_userpass)).split()
+    jazz_username, jazz_password = jazz_userpass_list[0], jazz_userpass_list[1]
+    jenkins_userpass_list = ''.join(list(jenkins_userpass)).split()
+    jenkins_username, jenkins_password = jenkins_userpass_list[0], jenkins_userpass_list[1]
+    apigee_userpass_list = ''.join(list(apigee_userpass)).split()
+    apigee_username, apigee_password = apigee_userpass_list[0], apigee_userpass_list[1]
+    install_proxy(
+        getTerraformOutputVar("apigee-lambda-user-secret-key"),
+        getTerraformOutputVar("apigee-lambda-user-id"),
+        region,
+        getTerraformOutputVar("apigee-lambda-gateway-func-arn"),
+        apigee_host,
+        apigee_org,
+        apigee_prod_env,
+        apigee_dev_env,
+        "1.0",
+        apigee_username,
+        apigee_password
+    )
+    credential_id = "ApigeeforJazz"
+    apigee_json = prepare_apigee_json(
+        apigee_host,
+        credential_id,
+        apigee_prod_env,
+        apigee_dev_env,
+        apigee_svc_prod_host,
+        apigee_svc_dev_host,
+        apigee_org
+    )
+
+    update_config(
+        "APIGEE",
+        apigee_json,
+        jazz_username,
+        jazz_password,
+        jazz_apiendpoint
+    )
+
+    # Store the CREDENTIAL_ID in jenkins
+    setCredential(jenkins_url, jenkins_username, jenkins_password, credential_id,
+                  apigee_username, apigee_password)
+    # Trigger metrics api
+    metricJobUrl = "job/build-pack-api/buildWithParameters?token=jazz-101-job&service_name=metrics&domain" \
+                   "=jazz&scm_branch=master"
+    startJob(jenkins_url, jenkins_username, jenkins_password, metricJobUrl)
+    # Trigger jazz ui
+    startJob(jenkins_url, jenkins_username, jenkins_password, "job/jazz_ui/build?token=jazz-101-job")
+
+
+@click.command()
+@click.option(
+    "--region",
+    "-r",
+    help='Specify the region where your Jazz installation lives',
+    type=click.Choice(['us-east-1', 'us-west-2']),
+    default='us-east-1'
+)
+@click.option(
+    "--stackprefix",
+    "-p",
+    help='Specify the stackprefix of your existing Jazz installation (e.g. myjazz), \
+          your existing config will be imported',
+    prompt=True
+)
+@click.option(
+    '--jazz_userpass',
+    nargs=2,
+    required=True,
+    help='Provide the username and password \
+    of the jazz application separated by a space',
+    prompt=True)
+@click.option(
+    "--jazz_apiendpoint",
+    help='Specify the Jazz Endpoint',
+    prompt=True
+)
+@click.option(
+    "--jenkins_url",
+    help='Specify the Jenkins url',
+    prompt=True
+)
+@click.option(
+    '--jenkins_userpass',
+    nargs=2,
+    required=True,
+    help='Provide the username and password \
+    of the jenkins separated by a space',
+    prompt=True)
+def uninstall(region, stackprefix, jazz_userpass, jazz_apiendpoint, jenkins_url, jenkins_userpass):
```

Comment:
Personal preference: I find it cleaner to keep the `click` variable declarations in a separate file from the logic that uses them, so there's a clean visual separation between the inputs and the execution. Makes everything easier to read.

I'd recommend moving the `uninstall/install/terraform` "do things" functions to their own file and referencing them from here instead.

Reply:
Done